File: app/src/main/java/com/chooloo/www/callmanager/ui/contact/ContactMvpPresenter.java
package com.chooloo.www.callmanager.ui.contact;

import com.chooloo.www.callmanager.ui.base.MvpPresenter;

public interface ContactMvpPresenter<V extends ContactMvpView> extends MvpPresenter<V> {
    void onBackButtonPressed();

    void onActionCall();

    void onActionSms();

    void onActionEdit();

    void onActionInfo();

    void onActionDelete();

    void onActionFav();

    void onRecentsLoadFinished();
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/contact/ContactMvpPresenter.kt
package com.chooloo.www.callmanager.ui.contact

import com.chooloo.www.callmanager.ui.base.MvpPresenter

interface ContactMvpPresenter<V : ContactMvpView> : MvpPresenter<V> {

    // contact actions
    fun onActionCall()
    fun onActionSms()
    fun onActionEdit()
    fun onActionInfo()
    fun onActionDelete()
}